### ISSUE TITLE
ForbiddenModule fails when alias __MODULE__.another_module is used

### ISSUE THREAD

**Issue by gordalina** [rrrene/credo#943]
### Environment

* Credo version (`mix credo -v`): `1.6.2`
* Erlang/Elixir version (`elixir -v`): `1.13.2`
* Operating system: `macOS 12.1 (arm64)`

### What were you trying to do?

Add a `ForbiddenModule` check

### Expected outcome

Expect it to work without failing

### Actual outcome

Credo exits with the following error:

```
[error] Task #PID<0.1987.0> started from #PID<0.1986.0> terminating
** (FunctionClauseError) no function clause matching in :elixir_aliases.do_concat/2
    (elixir 1.13.2) src/elixir_aliases.erl:152: :elixir_aliases.do_concat([{:__MODULE__, [line: 2, column: 9], nil}, :Module], "Elixir")
    (elixir 1.13.2) src/elixir_aliases.erl:140: :elixir_aliases.concat/1
    (credo 1.6.2) lib/credo/check/warning/forbidden_module.ex:33: Credo.Check.Warning.ForbiddenModule.traverse/4
    (elixir 1.13.2) lib/macro.ex:483: anonymous fn/4 in Macro.do_traverse_args/4
    (stdlib 3.16.1) lists.erl:1358: :lists.mapfoldl/3
    (elixir 1.13.2) lib/macro.ex:448: Macro.do_traverse/4
    (stdlib 3.16.1) lists.erl:1358: :lists.mapfoldl/3
    (elixir 1.13.2) lib/macro.ex:448: Macro.do_traverse/4
Function: #Function<4.65963795/0 in Credo.Check.Warning.ForbiddenModule.do_run_on_all_source_files/3>
    Args: []
```

### How to Replicate

1. Create a new project with `mix new credo_replicate_issue`
2. Add `{:credo, "1.6.2"}` as a dependency
2. Update the `credo_replicate_issue.ex` with the following content:
   ```ex
    defmodule CredoReplicateIssue do
      alias __MODULE__.Module

      def hello do
        Module.hello()
      end
    end

    defmodule CredoReplicateIssue.Module do
      def hello, do: :world
    end
    ```
3. Create a `.credo.exs` with the following config:
    ```ex    
    %{
      configs: [
        %{
          name: "default",
          files: %{included: ["lib/"]},
          checks: %{
            enabled: [
              {Credo.Check.Warning.ForbiddenModule, []}
            ],
          }
        }
      ]
    }
    ```
4. Run `mix deps.get && mix credo` and watch it fail

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by gordalina** [comment]
It works beautifully, thank you. Fixed by 2c95bac8237a679f8a90188342f6360e815813ce
